### PR TITLE
Pe-904 file with failed status red dot is shown as pending yellow dot

### PR DESCRIPTION
**Pull request by thiagocarvalhodev**
## Overview
This PR fixes the following issue:

When a file had failed to safely send its Data, i.e. Data Transaction, to ArWeave, this one wasn't updating its status to failed. So it remains pending when in fact it failed.

### Solution
We assume that the Data transaction was failed if we cannot get this one from Arweave after 24 hours since sent to ArWeave. So, we'll verify if the dateCreated of the transaction has more than 24 hours since its creation AND if we didn't have any response of this file DataTransaction from Arweave. If so, update the local database with `TransactionStatus.failed`.

#### Database
Add the `transactionDateCreated` to `NetworkTransactions`. With this date, we can verify if the Data Transaction failed.

Add a query `fileRevisionByDataTex`

#### SyncCubit
Add `_isOverThePedingTime` function.
Add `_getDateCreatedByDataTx` function.

### CODE REVIEW

File: lib/blocs/sync/sync_cubit.dart [hunk 1]
```diff
@@ -698,6 +716,26 @@ class SyncCubit extends Cubit<SyncState> {
     });
   }
 
+  bool _isOverThePedingTime(DateTime? transactionCreatedDate) {
```
Comment (fedellen) [inline lib/blocs/sync/sync_cubit.dart:719]: spelling nit:
```suggestion
  bool _isOverThePendingTime(DateTime? transactionCreatedDate) {
```

File: lib/blocs/sync/sync_cubit.dart [hunk 2]
```diff
@@ -698,6 +716,26 @@ class SyncCubit extends Cubit<SyncState> {
...
+    // If don't have the date information we cannot assume that is over the pending time
+    if (transactionCreatedDate == null) {
+      return false;
+    }
+
+    return DateTime.now().isAfter(transactionCreatedDate.add(_pendingWaitTime));
+  }
+
+  Future<DateTime?> _getDateCreatedByDataTx(String dataTx) async {
+    final rev = await _driveDao.fileRevisionByDataTex(tx: dataTx).get();
```
Comment (fedellen) [inline lib/blocs/sync/sync_cubit.dart:729]: This one is not actually part of your PR, but another spelling nit here:

```suggestion
    final rev = await _driveDao.fileRevisionByDataTx(tx: dataTx).get();
```

File: lib/models/tables/network_transactions.moor [hunk 3]
```diff
@@ -3,5 +3,7 @@ CREATE TABLE network_transactions (
 
     status TEXT NOT NULL DEFAULT 'pending',
 
-    dateCreated DATETIME NOT NULL DEFAULT (strftime('%s','now'))
+    dateCreated DATETIME NOT NULL DEFAULT (strftime('%s','now')),
+
+    transactionDateCreated DATETIME
 );
```
Comment (fedellen) [inline lib/models/tables/network_transactions.moor:9]: nit: Add new line at end of file

**Review by fedellen — COMMENTED** [review]
Looking good!

Left a few very minor comments

File: lib/blocs/sync/sync_cubit.dart (hunk 1 above)
Comment (thiagocarvalhodev) [reply lib/blocs/sync/sync_cubit.dart:719]: Thank you, @fedellen !

**Review by fedellen — APPROVED** [review]

**Review by karlprieb — APPROVED** [review]

**Review by jdaev — DISMISSED** [review]

**Review by jdaev — APPROVED** [review]

**Review by fedellen — APPROVED** [review]

**Pull request merged**